blog/2023-06-16-inline-sdk.mdx

```diff
@@ -0,0 +1,1104 @@
+```
+
+Open the folder with the name you supplied in VS Code. You should see a number of files already there.
+
+## Exploring the SDK
+
+There are a few key parts of the SDK that you can use to build your own custom band.
+
+### Registry Service
+
+Custom bands are registered in Inline using a custom VertiGIS web service. This service is found in `src/services/CustomBandRegistry/CustomBandRegistry.ts`. This file loads any available custom bands and registers them with Inline when it is added to an app. It can register any number of custom bands at the same time.
+
+:::warning
+Modifying this service is not recommended. It can result in unexpected behaviour including causing your app to crash. Do so at your own risk.
+:::
+
+### Types
+
+In the `src/definitions/inline.d.ts` file you can find all of the Inline types that you can use to create your custom band. There are comments provided in this file to describe the uses of these types.
+
+### Sample Band
+
+The Inline SDK comes preinstalled with a sample band that you can use for reference. In your SDK navigate to `src/bands/SampleBand`. In here you will find 4 files.
+
+-   `Band.css` contains the styles that are applied to the chart.
+-   `Band.tsx` contains the React code that is run in Inline.
+-   `designer.ts` describes the settings that are available when configuring a custom band in the Inline Designer.
+-   `index.ts` combines it all together to be properly registered.
+
+## Creating a new band
+
+The band we will be creating today will be able to display a video and seek to any point in the video based on the location of the crosshair on the map and on other inline charts.
+
+To create a new band run
+
+```
+npx @vertigis/inline-sdk generate VideoBand
+```
+
+:::important
+The `generate` command must be run from within the custom band project folder.
+:::
+
+### Add settings for the Inline Designer
+
+The `designer.ts` file provides the Inline designer with a settings schema to display configurable settings for a custom band. These settings can be any combination of:
+
+-   input
+-   checkbox
+-   radio
+-   dropdown
+
+The Sample Band has some examples of how to use these settings. The `src/bands/VideoBand/designer.ts` file will have a small function ready to go.
+
+```ts title="src/bands/VideoBand/designer.ts"
+import { InlineDesignerConfig } from "inline";
+
+export const getDesignerConfig = (): InlineDesignerConfig => {
+    const config: InlineDesignerConfig = [
+        {
+            title: "Video Band Settings",
+            fields: [],
+        },
+    ];
+
+    return config;
+};
+```
+
+The `InlineDesignerConfig` is a list of settings sections. Each section has a name and list of fields that can be configured by users in the Inline designer.
+
+Next, add an input field for the video url.
+
+```ts title="src/bands/VideoBand/designer.ts"
+// highlight-next-line
+import { InlineDesignerConfig, InputDesignerConfig } from "inline";
+
+export function getDesignerConfig() {
```

Comment:
The switch from `export const getDesignerConfig` to `export function getDesignerConfig()` will probably trip people up.

Reply:
Fixed.  Used the SDK default arrow function